### PR TITLE
Publications full text upload/download without versioning

### PR DESCRIPTION
**Pull request by abecam**
Used the fact that an asset can be non-versioned to do an adapted Upload. Implies some change in Seek Lib that shouldn't impact others assets. It is still not great, an asset can either have the normal functionality or this one, so a proper/separated upload functionality - more as an extra file for an element in Seek - might be preferred.
Other non-so great aspect is the fixed check for publication in the Asset partials to 1. remove the mandatory marker (the full text is pretty much non-mandatory), 2. print a warning message. 2. might be moved to the calling view in Publication, but probably not 1). It would be better to have a non-mandatory option in the model though.
Not sure if the version column in the publication table if of any use now, but I am not sure it is worth the risk.
The limits in the Publication table has been changed somehow during a migration... That bother me a bit but shouldn't affect anything.

### CODE REVIEW

File: lib/seek/assets_common.rb [hunk 1]
```diff
@@ -8,6 +8,11 @@ module AssetsCommon
     include Seek::AssetsStandardControllerActions
 
     def find_display_asset(asset = eval("@#{controller_name.singularize}"))
+      if !asset.versioned?
+        # No version, the display_ is the main asset, non-versionned
+        eval "@display_#{asset.class.name.underscore} = asset"
```
Comment (fbacall) [inline lib/seek/assets_common.rb:13]: generally `eval` should be avoided, even though there is no obvious security flaw in this case.

You could use `instance_variable_set("@display_#{asset.class.name.underscore}", asset)` instead.

But also see my above comment about compatibility for non-versioned things

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -247,7 +247,7 @@ def find_and_authorize_requested_item
       respond_to do |format|
         format.html do
           case privilege
-          when :publish, :manage, :edit, :download, :delete
+          when :publish, :manage, :edit, :download, :delete, :upload_fulltext
```
[the remark this answers is not in the document]
Comment (fbacall) [reply app/controllers/application_controller.rb:250]: I've seen later on that you did add the actions to the translator, so this line should be unnecessary.

[the remark this answers is not in the document]
Comment (abecam) [reply app/controllers/application_controller.rb:250]: Indeed, I added it later to the translator and did not think I didn't need it here anymore.

File: app/controllers/publications_controller.rb [hunk 3]
```diff
@@ -124,6 +133,26 @@ def update
     end
   end
 
+  def uploadPdf
```
[the remark this answers is not in the document]
Comment (abecam) [reply app/controllers/publications_controller.rb:136]: It was underlined. I probably need to change the color, was too light for me.

File: app/views/assets/_fileinfo.html.erb [hunk 4]
```diff
@@ -1,41 +1,46 @@
 <div class='fileinfo'>
-<% if fileinfo.show_as_external_link?  -%>
-    <% if fileinfo.asset.can_download? %>
+  <% if fileinfo != nil -%>
```
[the remark this answers is not in the document]
Comment (abecam) [reply app/views/assets/_fileinfo.html.erb:2]: Changed it. Not used to have such condition :)

File: app/controllers/publications_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (abecam) [reply app/controllers/publications_controller.rb:136]: And changed (but rubocop still complains about some other things, I will check)

File: lib/seek/assets_common.rb (hunk 1 above)
Comment (abecam) [reply lib/seek/assets_common.rb]: Changed it anyway, also the 2nd one which was there before :)

File: app/views/assets/_upload.html.erb [hunk 5]
```diff
@@ -20,6 +20,9 @@ end
     <p>
       You can register a <%= asset_name -%> by either directly uploading a file,
        or registering a URL to either another page or remote file. Please test the URL before submitting.
+      <% if (controller_name.eql? "publications")%>
```
[the remark this answers is not in the document]
Comment (abecam) [reply app/views/assets/_upload.html.erb:23]: Nicer indeed, always good to get rid of string literal, and changed.

**Comment by abecam** [comment]
Will do a new request when the new soft delete is implemented (testing now, probably need some tuning due to the specific access rights of the publications)

**Pull request closed without merging**